src/lib/vocab/entity/entityHelpers.ts

```diff
@@ -50,3 +51,24 @@ export const checkEntityPath = (path: string): string | null => {
 	}
 	return null;
 };
+
+/**
+ * Takes an entity with orderedItems in its data
+ * and constructs an ordered array of its items
+ * @param orderedCollection
+ * @param ui
+ * @returns null if invalid, otherwise an array of readable entities
+ */
+export const constructOrderedCollection = (
+	orderedCollection: Entity,
+	ui: Ui,
+): Readable<Array<Readable<Entity>>> | null => {
+	//TODO add recovery logic i.e.
+	//if tere are entities in collection, but not orderedIteam array, append
+	//or if there are ids in the array, but not tied to collection; remove them
+	const {entityById} = ui;
+	if (!('orderedItems' in orderedCollection.data)) return null;
+	if (!orderedCollection.data.orderedItems) return null;
```

Comment:
looks like this is redundant, and isn't needed for type narrowing:

```suggestion
	if (!orderedCollection.data.orderedItems) return null;
```